newrelic/data_source_newrelic_synthetics_private_location.go

```diff
@@ -77,5 +84,27 @@ func dataSourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *sch
 		return diag.FromErr(err)
 	}
 
+	key := fetchPrivateLocationKey(location.Tags)
+	if len(key) == 0 {
+		//logs the absence of a key but does not throw an error to prevent halting execution
+		log.Printf("[INFO] No keys found corresponding to the queried private location.")
+	}
+	err = d.Set("key", key)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
 	return nil
 }
+
+// fetchPrivateLocationKey extracts the 'key' of the private location by iterating through all tags
+// and finding the tag by the name 'key', the 'values' of which would contain the key of the private location
+func fetchPrivateLocationKey(tagList []entities.EntityTag) []string {
+	var key []string
+	for _, tag := range tagList {
+		if tag.Key == "key" {
+			key = tag.Values
+		}
+	}
+	return key
+}
```

Comment:
For the reviewer: a brief explanation of what's being done here - 

The response of the NG query would contain the **key** of the private location in the list of tags returned, so this function extracts the tag by name **key** and returns its values. Though only one string is supposed to be returned, in order to maintain consistency with the datatype returned by NG, a list of strings (with only one string) is being returned.